client.go

```diff
@@ -230,8 +229,10 @@ func (c *Client) Close() error {
 	if c.cancelMonitor != nil {
 		c.cancelMonitor()
 	}
-
-	return c.sechan.Close()
+	if c.sechan != nil {
+		return c.sechan.Close()
+	}
+	return io.EOF
```

Comment:
you return `ua.StatusBadServerNotConnected` everywhere else `sechan` is nil. Any reason to deviate here?

Reply:
I reckon, when one attempts to close a `Client` and there's no `sechan` to close, we can assume that everything is closed and return `io.EOF`, which is what `sechan` would return to signal a successful closure.